packages/flutter/lib/src/widgets/router.dart

```diff
@@ -447,62 +511,33 @@ class _RouterState<T> extends State<Router<T>> {
     assert(_routeInformationReportingTaskScheduled);
     _routeInformationReportingTaskScheduled = false;
 
-    switch (_currentIntentionToReport) {
-      case _IntentionToReportRouteInformation.none:
-        assert(false);
-        return;
-
-      case _IntentionToReportRouteInformation.ignore:
-        // In the ignore case, we still want to update the _lastSeenLocation.
-        final RouteInformation? routeInformation = _retrieveNewRouteInformation();
-        if (routeInformation != null) {
-          _lastSeenLocation = routeInformation.location;
-        }
-        _currentIntentionToReport = _IntentionToReportRouteInformation.none;
-        return;
-
-      case _IntentionToReportRouteInformation.maybe:
-        final RouteInformation? routeInformation = _retrieveNewRouteInformation();
-        if (routeInformation != null) {
+    if (_routeInformation.value != null) {
+      final RouteInformation routeInformation = _routeInformation.value!;
+      switch (_currentIntentionToReport) {
+        case _IntentionToReportRouteInformation.none:
+          assert(false);
```

Comment:
Nit: Maybe add a string parameter here to describe what's wrong to the developer.

Reply:
Added a message. When this assert fires it will indicate a bug in the framework, though.